core/trie/trie.go

```diff
@@ -13,7 +13,7 @@ import (
 	"github.com/NethermindEth/juno/db"
 )
 
-type hashFunc func(*felt.Felt, *felt.Felt) *felt.Felt
+type HashFunc func(*felt.Felt, *felt.Felt) *felt.Felt
```

Comment:
Why is thus public? Can we not keep it private?

Reply:
I think you can keep this function private, but expose it via t.HashFunc()?

Reply:
Yeah, when I added:
`func (t *Trie) HashFunc() HashFunc {`
compiler wanted me to expose it's return type public

Reply:
It's not giving me any compilation problems (see [here](https://github.com/NethermindEth/juno/commit/d51bc26601c0c388b6d1343658a0da09ddca5460)), but feel free to ignore me if I'm doing something wrong :)

Reply:
Maybe it wasn't compiler but linter. Anyway I'm leaving as is - publicly exposed function uses public type.
Thanks for the suggestions 👍